os/os.proto

```diff
@@ -120,12 +132,18 @@ service OS {
   rpc Verify(VerifyRequest) returns (VerifyResponse);
 }
 
+// Systems which save image and config to secondary disk on the same Supervisor for
+// auto recovery from failed upgrade will snapshot to secondary disk if snapshot_to_secondary_disk
+// is true
 message InstallRequest {
   oneof request {
     TransferRequest transfer_request = 1;
     bytes transfer_content = 2;
     TransferEnd transfer_end = 3;
   }
+  // For auto-recovery, snapshot will save the image and config to secondary disk 
+  // on the same Supervisor.
+  bool snapshot_to_secondary_disk = 4;
```

Comment:
Current image and config? This should happen regardless, so this flag is not needed. See lines 44 and 366.